components/ProductForm.js

```diff
@@ -1,6 +1,11 @@
 import useSWR from "swr";
 
-export default function ProductForm({ onSubmit }) {
+export default function ProductForm({
+  onSubmit,
+  defaultData = {},
+  submitLabel = " Add new product",
+  onCancel,
+}) {
   const { data, isLoading, error } = useSWR("/api/categories");
```

Comment:
please rename this to categories, this helps in the future to prevent confusion.